articles/multifactor-authentication/mfa-from-id-token.md

```diff
@@ -0,0 +1,28 @@
+---
+description: Describes how to determine if a user has done multifactor authentication using their id_token and JWT
+---
+
+## Determine if a user has performed multifactor authentication
+
+At times, it's necessary to determine if a particular user has had the additional security of multifactor authentication applied to their session. For instance, a user may be allowed access to sensitive data or allowed to reset their password only after further confirming their identity using MFA.
+
+To determine this for a particular user session, developers can check the claims information available on the `id_token`. 
+The `id_token` contains claims information relevant to the usage of multifactor authentication. After retrieving the `id_token`, [as shown here](tokens/id_token), the value of the `amr` claim can be evaluated.
+
+```
```

Comment:
I know that I sent you the example, but it has some small mistakes, here is an examples that actually works:

```js
const AUTH0_CLIENT_SECRET = '${account.clientSecret}';
const jwt = require('jsonwebtoken')

jwt.verify(id_token, new Buffer(AUTH0_CLIENT_SECRET, 'base64'), { algorithms: ['HS256'] }, function(err, decoded) {
   if (err) {
     console.log('invalid token');
     return;
   }
 
   if (Array.isArray(decoded.amr) && decoded.amr.indexOf('mfa') >= 0) {
     console.log('You used mfa');
     return;
   }
 
   console.log('you are not using mfa');
 });
```